docs/tutorials/agent_framework/Chatbot_with_RAG.md

```diff
@@ -0,0 +1,318 @@
+
+```
+## 1. Create LLM
+
+Follow step 2 "Prepare LLM" of [RAG_with_conversational_flow_agent](./RAG_with_conversational_flow_agent.md) to set up Bedrock Claude model.
+
+Note the model id, you will use it in following steps.
+
+## 2. Create Agent
+Create agent with `conversational` type. 
+
+Both `conversational_flow` and `conversational` agents can support conversation history.
+
+Difference:
+- `conversational_flow` agent runs tools sequentially with predefined order.
+- `conversational` agent dynamically chooses which tool to run.
+
+This agent includes two tools which provides recent population data and tech news.
+
+Explanation:
+
+- `"max_iteration": 5`: Agent will run at most 5 times of LLM.
+- `"response_filter": "$.completion"` is to filter only the answer from Bedrock Claude model response
+- `"doc_size": 3` in `population_data_knowledge_base`: means it will return top 3 documents.
+```
+POST _plugins/_ml/agents/_register
+{
+  "name": "Chat Agent with RAG",
+  "type": "conversational",
+  "description": "this is a test agent",
+  "llm": {
+    "model_id": "your_llm_model_id",
+    "parameters": {
+      "max_iteration": 5,
+      "response_filter": "$.completion"
+    }
+  },
+  "memory": {
+    "type": "conversation_index"
+  },
+  "tools": [
+    {
+      "type": "VectorDBTool",
+      "name": "population_data_knowledge_base",
+      "description": "This tool provides population data of US cities.",
+      "parameters": {
+        "input": "${parameters.question}",
+        "index": "test_population_data",
+        "source_field": [
+          "population_description"
+        ],
+        "model_id": "your_text_embedding_model_id",
+        "embedding_field": "population_description_embedding",
+        "doc_size": 3
+      }
+    },
+    {
+      "type": "VectorDBTool",
+      "name": "tech_news_knowledge_base",
+      "description": "This tool provides recent tech news.",
+      "parameters": {
+        "input": "${parameters.question}",
+        "index": "test_tech_news",
+        "source_field": [
+          "passage"
+        ],
+        "model_id": "your_text_embedding_model_id",
+        "embedding_field": "passage_embedding",
+        "doc_size": 2
+      }
+    }
+  ],
+  "app_type": "chat_with_rag"
+}
+```
+Note the agent id, you will use it in next step.
+
+## 3. Test Agent
+
+`conversational` agent supports `verbose`. You can set it as `true` to see detail steps.
+
+Or you can use get trace data API.
+```
+GET _plugins/_ml/memory/message/message_id/traces
+```
+
+### 3.1 Start new conversation
+- Example1: ask question related to tech news
+```
+POST _plugins/_ml/agents/your_agent_id/_execute
+{
+  "parameters": {
+    "question": "What's vision pro",
+    "verbose": true
+  }
+}
+```
+From response, we can see Agent runs `tech_news_knowledge_base` tool and get top 2 documents. Agent will pass these documents as context to LLM.
```

Comment:
From the response, we can see the Agent runs the `tech_news_knowledge_base` tool and retrieves the top 2 documents. The Agent then passes these documents as context to the LLM, allowing it to accurately answer the question.